### ISSUE TITLE
Container starts ramping up CPU & mem usage

### ISSUE THREAD

**Issue by Manuel5cc** [ThePhaseless/Byparr#14]
I had this container running for several days and it works much better than the original flaresolver, but if left running "unattended", it starts eating up resources. 

These graphs show the cpu and memory evolution until I stopped it around 7 am, where you can clearly see that the resources are freed.

![rrd](https://github.com/user-attachments/assets/2d71a9c5-3b84-4ef4-92c9-9f349964cd68)
![rrd (1)](https://github.com/user-attachments/assets/00b8c1f9-5ad7-44b4-a0f6-5df426520c0d)

**Comment by ThePhaseless** [comment]
Hi, are you sure that you're using the latest version? There was a moment where a bug prevented the browser to close after failure

**Comment by Manuel5cc** [comment]
> Hi, are you sure that you're using the latest version? There was a moment where a bug prevented the browser to close after failure

I pulled the :latest version so it should.

**Comment by ThePhaseless** [comment]
Hi, I've added a hotfix that ends browser on every exception, should fix your problem, please pull now

**Comment by Manuel5cc** [comment]
Hi there!
This started happening again since a few days ago. I normally pull new images on saturday so I guess if there was a new image in the last 15 days that's causing the issue.

**Comment by ThePhaseless** [comment]
That is strange... After the rework, now the implementation implies that everything closes after the request is finished (whether it was successful or not), but I've added a driver quit anyways. Please check if version v1.0.4 fixes this problem

**Comment by Manuel5cc** [comment]
> That is strange... After the rework, now the implementation implies that everything closes after the request is finished (whether it was successful or not), but I've added a driver quit anyways. Please check if version v1.0.4 fixes this problem

It has started happening again after 8 hours running with the new release. The logs don't show anything out of the ordinary happening at the same time or during the high CPU periods. For now I found a workaround, I restart the container every hour.

**Comment by ThePhaseless** [comment]
Hi, I think I know what's happening. Sometimes chrome can crash at startup (I think). So there is no way to directly catch this. I've added a function that will kill all chrome processes that are longer than 5 minutes at health check endpoint (which is triggered with docker healthcheck each 5 mins). Currently this is only on `:latest` tag. I will create a release after testing

**Comment by ThePhaseless** [comment]
As I do not see a problem on my site anymore and no responses here, I'll close this

**Comment by arlexTech** [comment]
i have the issue here, byparr is taking a very huge amount of cpu 

CONTAINER ID   NAME                                    CPU %     MEM USAGE / LIMIT     MEM %     NET I/O           BLOCK I/O         PIDS
7880f0a6fc8d   byparr                                  111.55%   1.056GiB / 31.23GiB   3.38%     2.94GB / 443MB    5.92GB / 54.2GB   633

**Comment by Shocco** [comment]
Im also seeing this after several hours. possible regression?

**Comment by jhou322** [comment]
I'm also seeing this

**Comment by mdeeter** [comment]
I'm seeing this as well... regularly ends up at almost 200% cpu after 16 hours of uptime

**Comment by poiromaniax** [comment]
Same here. @ThePhaseless FYI - not sure if youre seeing these because the issue is closed

<img width="1013" height="110" alt="Image" src="https://github.com/user-attachments/assets/234763d9-48d0-4fe1-94e9-040b006b40f4" />

**Comment by ThePhaseless** [comment]
Yeah, if you ping me I'll see it. I will try to rewrite it with Camoufox. Chrome is unstable af

**Comment by poiromaniax** [comment]
Thanks :)

**Comment by Xavi-Merino** [comment]
Hi, this is happening also to me. is there a way i could contribute with logs or a test to solve this?

**Comment by Jacco011** [comment]
It uses 10GB of memory on Debian12 but CPU is almost 0. Running the latest stable.

**Comment by ThePhaseless** [comment]
@Xavi-Merino @Jacco011 could you please provide your compose configs and 'top' command output with sorting by memory descending?
